Trivial fixes for MP3Filter

bertm opened the pull request:
This fixes a few header parsing and consistency bugs in the MP3Filter, while introducing support for dropping entire ID3v1 and ID3v2 tags.

I tested this with a random collection of music files on my PC, as well as @ArneBab's reported broken file, and all were filtered well where they were previously all broken.

ArneBab commented:
I don’t understand mp3 well enough to really judge the pull-request, but it looks good, and it definitely shouldn’t make anything worse.

I plan to test this in a bit more than a week.

Bombe commented on src/freenet/client/filter/MP3Filter.java:
```diff
@@ -118,9 +118,12 @@ public void filter(InputStream input, OutputStream output) throws DataFilterExce
+					foundStream = false;
+					continue; // Not valid
+				}
+				boolean paddingBit = ((frameHeader & 0x00000200) >>> 9) == 1 ? true : false;
```
`? true : false`? Seriously? Please remove that. :)

Bombe commented on src/freenet/client/filter/MP3Filter.java:
```diff
@@ -167,6 +170,29 @@ public void filter(InputStream input, OutputStream output) throws DataFilterExce
+				Logger.normal(this, "Skipped " + size + " bytes of ID3v2 data");
+				frameHeader = in.readInt();
+				foundStream = (frameHeader & 0xffe00000) == 0xffe00000;
+			} else if (!foundStream && (frameHeader & 0xffffff00) == 0x54414700) {
```
ID3v1 headers are not located at the beginning of the file. Am I missing something here?

Bombe commented:
The ID3 tag skipping looks good, too. Now the only thing missing is some refactoring and a test. :)

ArneBab commented:
Feel free to use my file. The song is a traditional and it’s sung by me.

However 500kiB could be a bit too large…

Bombe commented:
Uhm, encode a second of silence and tag that. No need for more than that.

bertm commented:
I'm working on it.
Op 12 jan. 2016 19:31 schreef "David Roden" <notifications@github.com>:

> Uhm, encode a second of silence and tag that. No need for more than that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freenet/fred/pull/478#issuecomment-171004038>.
>

Bombe commented:
@bertm awesome!

ArneBab commented:
Here’s a one second file with a dieing sine sound: CHK@x8~GoW0UAZ9PR8W44XTFN3-U6IWET3DP3yHqwylsGFI,m62xSWGyvgSviCj562RXB7exZTcPXgDBEjNzXeX5zuQ,AAMC--8/dieing-sine-48khz.mp3

(no silence to ensure that we *hear* it when the filter breaks)

ArneBab commented:
@bertm will you finish this yourself? I’m working on an m3u filter right now #510 so along with #429 we’d have full support for radio over Freenet.

bertm commented:
Yep, I have something almost commit-ready laying around. My thesis sure is a killer to my available time, but I'll try and have a final look at it tonight.

Thanks for your work on an M3U filter by the way!

ArneBab commented:
thank you!

bertm commented:
I added several test instances of 200ms worth of generated audio data, encoded in various ways. I did not choose to use silence, for that would reduce the variability of the MPEG frames to approximately zero.

Implementing this I encountered and fixed yet another bug (this time caused by partial or even erroneous online documentation of the peculiarities of frame length).

bertm commented:
The integration test failure is completely unrelated; one of the tests ran out of heap space in some splitfile test.
